### PR TITLE
luci-app-samba4: add new UCI option "enable_extra_tuning"

### PR DESCRIPTION
**Pull request by Andy2244**
* add new UCI option "enable_extra_tuning"

Depends on openwrt/packages#12630

### CODE REVIEW

File: applications/luci-app-samba4/htdocs/luci-static/resources/view/samba4.js [hunk 1]
```diff
@@ -39,13 +39,17 @@ return view.extend({
...
 		o.placeholder = 'Samba4 on OpenWrt';
 		
+		s.taboption('general', form.Flag, 'enable_extra_tuning', _('Enable extra Tuning'),
+			_('Enable some community driven tuning parameters, that may improve write speeds and better operation via WiFi.\
+			Not recommend if multiple clients write to the same files, at the same time!'));
+		
 		s.taboption('general', form.Flag, 'disable_async_io', _('Force synchronous  I/O'),
 			_('On lower-end devices may increase speeds, by forceing synchronous I/O instead of the default asynchronous.'));
 
-		o = s.taboption('general', form.Flag, 'macos', _('Enable macOS compatible shares'),
```
Comment (feckert) [inline applications/luci-app-samba4/htdocs/luci-static/resources/view/samba4.js:45]: Why did you changed that as well?
This does not match your commit message

File: applications/luci-app-samba4/htdocs/luci-static/resources/view/samba4.js [hunk 2]
```diff
@@ -39,13 +39,17 @@ return view.extend({
...
+		s.taboption('general', form.Flag, 'macos', _('Enable macOS compatible shares'),
 			_('Enables Apple\'s AAPL extension globally and adds macOS compatibility options to all shares.'));
 			
-		o = s.taboption('general', form.Flag, 'allow_legacy_protocols', _('Allow legacy (insecure) protocols/authentication.'),
```
Comment (feckert) [inline applications/luci-app-samba4/htdocs/luci-static/resources/view/samba4.js:48]: The same there

File: applications/luci-app-samba4/htdocs/luci-static/resources/view/samba4.js (hunk 1 above)
Comment (Andy2244) [reply applications/luci-app-samba4/htdocs/luci-static/resources/view/samba4.js:45]: its just a minor change, the assignment is not needed, thats all

**Comment by feckert** [comment]
**Note:**
Normally every change should be packed into a separate commit.
Then these changes could be better cherry-picked.
Please keep this in mind in the future when you create a pullrequest.
Any change that does something different and is not too concatenated should be put into a separate commit.

I will merge this now.
Thanks for your contribution

**Pull request merged**